src/main/java/jchess/ruleengine/DefaultRuleProcessor.java

```diff
@@ -432,4 +432,97 @@ public IPlayerAgent tryCheckIfPlayerEndengered(IBoardAgent oBoard, IPlayerAgent
 				return null;	
 	}
 
+	/**
+	 * Checking for Stalemate Rule
+	 * 
+	 * Stalemate is a situation in the game of chess where the player whose turn 
+	 * it is to move is not in check but has no legal move. 
+	 * The rules of chess provide that when stalemate occurs, the game ends as a draw.
+	 */
+	
+	public Boolean checkStalemate(IBoardAgent oBoard, IPlayerAgent oPlayer) {
+		Boolean bStalemateCheckIfPlayerEndengered = false;
+		Boolean bStalemateCheckIfPlayNotPossible = false;
+		int nCountNoOfPieces =0;
+		int nCountPiecesTrapped=0;
+		int nCounterOfKingMoves=0;
+		int[] iaArrayOfPossibleChecks = new int[]{0,0,0,0,0,0,0,0,0}; 
+		//fetching all pieces from the board
+		for(Map.Entry<String,IPositionAgent> oPositionPiece : oBoard.getAllPositionAgents().entrySet()) 
+		{
+			IPieceAgent oRandomPiece = oPositionPiece.getValue().getPiece();
+			//check to filter out only the pieces of the current player
+			if(oRandomPiece != null && oRandomPiece.getPlayer()== oPlayer) 
+			{
+				nCountNoOfPieces+=1;
+				//checking if the piece iterated is a king
+				if(oRandomPiece.getPieceData().getName().startsWith("King")) 
```

Comment:
1. You can directly get name from oRandomPiece.getName().
2. Danny has added a function in it. Try oPlayer.getKingPiece()

Reply:
Changed it to oRandomPiece.getName().